src/quick-lint-js/fe/lex.cpp

```diff
@@ -996,7 +996,12 @@ void Lexer::skip_less_less_as_less() {
 void Lexer::skip_as_greater() {
   switch (this->last_token_.type) {
   case Token_Type::greater_equal:
-    this->last_token_.type = Token_Type::equal;
+    if (this->input_[0] == '>') {
+      this->last_token_.type = Token_Type::equal_greater;
```

Comment:
Must fix: The `=>` token's .end is incorrect after this change. The following example demonstrates the bug:
```typescript
class C<T> {
  f(): C<T>=> {
  }
}
```

Reply:
Fixed.  What do you think?